### PR TITLE
Attempt to fix a crash when roomSummary is `nil` in filter results

### PR DESCRIPTION
**Pull request by matrixuser0**
Attempted fix for #7403

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [ ] Pull request contains a changelog file in ./changelog.d. See https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#changelog
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#sign-off)

### CODE REVIEW

**Comment by Velin92** [comment]
Hi thanks a lot for the contribution and the fix! That's great! @matrixuser0  if you could include the signoff and the changelog too, I will approve it!

**Comment by Velin92** [comment]
This PR fixes: https://github.com/vector-im/element-ios/issues/7403

File: MatrixSDK/Data/RoomList/Common/MXBreadcrumbsRoomListDataFetcher.swift [hunk 1]
```diff
@@ -91,7 +91,7 @@ internal class MXBreadcrumbsRoomListDataFetcher: NSObject, MXRoomListDataFetcher
         
         if let query = fetchOptions.filterOptions.query?.lowercased(), !query.isEmpty {
             recentRoomIds = recentRoomIds.filter({ roomId in
-                guard let displayname = session?.roomSummary(withRoomId: roomId).displayname else {
+                guard let displayname = session?.roomSummary(withRoomId: roomId)?.displayname else {
```
Comment (Velin92) [inline MatrixSDK/Data/RoomList/Common/MXBreadcrumbsRoomListDataFetcher.swift:94]: I would add a comment at the top this line to remind that the function `roomSummary(withRoomId: )` might return a nil value.
In this way the issue should not happen again.

**Comment by dfnik** [comment]
@Velin92 this is an intentionally anonymous account, so feel free to re-create with proper sign-off and Changelog

**Comment by Velin92** [comment]
Thanks I totally understand! I will recreate the PR again and close this one. 
Thanks again for your help

**Comment by matrixuser0** [comment]
@Velin92 since this is an anonymous account, I don't think I can create proper sign-off. So please feel free to create a new PR with your own sign-off.

**Pull request closed without merging**